core/modules/node/node.types.inc

```diff
@@ -661,11 +661,13 @@ function node_type_form_permissions($type) {
     );
     foreach ($roles as $role_name => $role) {
       // Builds arrays for checked boxes for each role. For new content types,
-      // if user_admin_role and/or user_editor_role have been configured in the
-      // "Default roles" section in admin/config/people/roles, then always
-      // select all permissions for these roles.
+      // if user_admin_role and/or user_editor_role have been configured, then
+      // select sensible permissions for these roles.
       if (in_array($perm, $role->permissions) || (!$type && ($role->name == $admin_role || $role->name == $editor_role))) {
-        $status[$role_name][] = $perm;
+        // Do not allow editors to delete any content.
+        if (!($role->name == $editor_role && strstr($perm, 'delete any'))) {
```

Comment:
These here are the "placeholder" permissions (without the content type name in their name), because the conditional after the OR has `!$type` (which means "this is a new content type"). So it should be safe to use the full name of that permission:
```suggestion
        if (!($role->name == $editor_role && $perm == 'delete any content')) {
```